keras_trainer/trainer.py

```diff
@@ -58,40 +69,73 @@ def run(self):
             fill_mode='nearest'
         )
 
-        val_generator = self.val_generator or image.ImageDataGenerator(
-            preprocessing_function=self.model_spec.preprocess_input
-        )
-
-        train_gen = train_generator.flow_from_directory(
+        train_gen = self.train_generator or train_data_generator.flow_from_directory(
             self.train_dataset_dir,
             batch_size=self.batch_size,
             target_size=self.model_spec.target_size[:2],
             class_mode='categorical'
         )
 
-        val_gen = val_generator.flow_from_directory(
+        # Set up the validation data generator.
+        val_data_generator = self.val_data_generator or image.ImageDataGenerator(
+            preprocessing_function=self.model_spec.preprocess_input
+        )
+
+        val_gen = self.val_generator or val_data_generator.flow_from_directory(
             self.val_dataset_dir,
             batch_size=self.batch_size,
             target_size=self.model_spec.target_size[:2],
             class_mode='categorical',
             shuffle=False
         )
 
-        model = self.model_spec.klass(
-            self.model_spec.target_size,
-            alpha=self.alpha,
-            weights=self.weights,
-            include_top=False,
-            pooling=self.pooling
-        )
-
-        if self.top_layers is None:
-            layer = Dense(len(self.dictionary), name='dense')(model.output)
-            self.top_layers = Activation('softmax', name='act_softmax')(layer)
-        model = Model(model.input, self.top_layers)
-
-        model.summary()
-
+        # Initialize the model instance.
+        if self.checkpoint_path is not None:
+            model = load_model(self.checkpoint_path)
```

Comment:
Will we want to load from a checkpoint and also expand for a new target size?

Reply:
I think it only has sense the first time you begin training the model.